edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/deployment/KubernetesDeploymentMapper.cs

```diff
@@ -302,7 +302,7 @@ List<V1EnvVar> CollectModuleEnv(KubernetesModule module, IModuleIdentity identit
                     this.agentConfigVolume.HasValue)
                 {
                     var agentConfigVolume = this.agentConfigVolume.OrDefault();
-                    volumeList.Add(new V1Volume { Name = agentConfigVolume, ConfigMap = new V1ConfigMapVolumeSource(name: this.agentConfigMapName.OrDefault()) });
+                    volumeList.Add(new V1Volume { Name = agentConfigVolume, ConfigMap = new V1ConfigMapVolumeSource(name: this.agentConfigMapName.OrDefault(), optional: true) });
```

Comment:
What does this mean?

Reply:
This allows Agent to start if the configmap is not present.  I had done this in iotedged as well, but missed it in the c# code.  While we don't allow currently allow Agent to update without updating the Helm chart, I wanted to make sure this didn't break/crash in case our users used beta11 with beta 10 helm charts.